core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1552,30 +1560,45 @@ class KafkaApis(val requestChannel: RequestChannel,
         sendResponseMaybeThrottle(request, requestThrottleMs =>
           new AddPartitionsToTxnResponse(requestThrottleMs, partitionErrors.asJava))
       } else {
-        // Send response callback
-        def sendResponseCallback(error: Errors): Unit = {
-          def createResponse(requestThrottleMs: Int): AbstractResponse = {
-            val responseBody: AddPartitionsToTxnResponse = new AddPartitionsToTxnResponse(requestThrottleMs,
-              partitionsToAdd.asScala.map{tp => (tp, error)}.toMap.asJava)
-            trace(s"Completed $transactionalId's AddPartitionsToTxnRequest with partitions $partitionsToAdd: errors: $error from client ${request.header.clientId}")
-            responseBody
+        val (_, incorrectMessageFormat) = partitionsToAdd.asScala.partition { tp =>
+          replicaManager.getMagic(tp) match {
```

Comment:
My assumption is that if the topic doesn't support the message format, then the producer will be doomed to abort as soon as they try to write to it. If they go ahead and commit anyway (maybe because of a client bug), I still think it's "ok" in the sense that no transactional data could have been written to the topic which didn't support the message format, so there will be no need to write any markers (committing is a no-op in some sense). This is similar to the problem with topic deletion, which could occur after we have accepted a commit.